graphql/src/test/groovy/io/micronaut/configuration/graphql/GraphQLControllerSpec.groovy

```diff
@@ -68,6 +77,16 @@ class GraphQLControllerSpec extends Specification {
         executionInput = null
         1 * graphQL.executeAsync(_) >> { ExecutionInput executionInput ->
             this.executionInput = executionInput
+            if (executionInput.query == "{ testHeaders }") {
+                GraphQLContext graphQlContext = executionInput.getContext()
```

Comment:
I think these expectations/asserts should be moved to the particular test (so no need to test on the query "testHeaders" here .

Reply:
The problem here is this is in `setup` method already, I mean that assert:

```groovy
1 * graphQL.executeAsync(_) >> { ExecutionInput executionInput ->
```

I do not very understand how to move my additional changes away from here, because this `setup` method will be executed for each test, if I remove my changes from here, then I will get the error from my test:

```groovy
Too few invocations for:

1 * graphQL.executeAsync(_) >> { ExecutionInput executionInput ->
            this.executionInput = executionInput
            return executionResult
        }   (0 invocations)
```

Because that setup method will be executed for my test as well.

I can see a few options here:

1) Move my test to a separate class
2) Add checking for current test name in the setup method by using `specificationContext.currentIteration.name`, which is for me is a hack-way and doesn't seem to be a correct way
3) Rename `setup` method and for each, not my test add invoking that renamed method

What do you think, maybe you have a better suggestion?